examples/xhr-proxy/xhrproxy.js

```diff
@@ -0,0 +1,55 @@
+/* This is an example how to monkey-patch XMLHttpRequest() for HTTP requests to a local service port.
+ * Cockpit pages cannot use the standard XMLHttpRequest() due to its strict Content-Security-Policy,
+ * so these need to be wrapped into a raw cockpit channel and be "really" done from the host where
+ * the session runs.
```

Comment:
I'd include text like "This is useful if you're wrapping existing javascript code that uses XMLHttpRequest(), and you're unable to change that code to use cockpit.http()."